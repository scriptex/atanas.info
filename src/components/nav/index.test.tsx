import * as React from 'react';
import { shallow } from 'enzyme';

import { Nav } from '.';

describe('Nav component', () => {
	it('Should render the Nav component', () => {
		const wrapper = shallow(<Nav />);

		expect(wrapper).toMatchSnapshot();
	});

	it('Should render the Nav component with props', () => {
		const wrapper = shallow(
			<Nav hasShell={true} className="c-nav--test" onClick={jest.fn()}>
				Test
			</Nav>
		);

		expect(wrapper).toMatchSnapshot();
	});
});
