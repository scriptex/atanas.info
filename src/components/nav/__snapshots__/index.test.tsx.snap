// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavComponent component Should render the NavComponent component 1`] = `
<DocumentFragment>
  <nav
    class="c-nav c-nav--test"
  >
    <div
      class="o-shell"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Back to homepage"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class=""
            href="/about"
            title="About me"
          >
            About
          </a>
        </li>
        <li>
          <a
            class=""
            href="/timeline"
            title="Life events"
          >
            Timeline
          </a>
        </li>
        <li>
          <a
            class=""
            href="/skills"
            title="My skills"
          >
            Skills
          </a>
        </li>
        <li>
          <a
            class=""
            href="/portfolio"
            title="My portfolio"
          >
            Portfolio
          </a>
          <ul>
            <li>
              <a
                class=""
                href="/portfolio/web-applications"
                title="My web applications portfolio"
              >
                Web applications
              </a>
            </li>
            <li>
              <a
                class=""
                href="/portfolio/mobile-applications"
                title="My mobile applications portfolio"
              >
                Mobile applications
              </a>
            </li>
            <li>
              <a
                class=""
                href="/portfolio/automotive-applications"
                title="My automotive applications portfolio"
              >
                Automotive applications
              </a>
            </li>
          </ul>
        </li>
        <li>
          <a
            class=""
            href="/stats"
            title="My stats"
          >
            Stats
          </a>
        </li>
        <li>
          <a
            class=""
            href="/slides"
            title="My presentations"
          >
            Slides
          </a>
        </li>
        <li>
          <a
            class=""
            href="/videos"
            title="My video presentations"
          >
            Videos
          </a>
        </li>
        <li>
          <a
            class=""
            href="/articles"
            title="My published articles"
          >
            Articles
          </a>
        </li>
        <li>
          <a
            class=""
            href="/certificates"
            title="My certificates"
          >
            Certificates
          </a>
        </li>
        <li>
          <a
            class=""
            href="/social"
            title="Social widgets"
          >
            Social
          </a>
        </li>
        <li>
          <a
            class=""
            href="/blog"
            title="My blog"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            class=""
            href="/resume"
            title="My resume"
          >
            Resume
          </a>
        </li>
        <li>
          <a
            class=""
            href="/interactive-resume"
            title="My interactive resume"
          >
            Interactive resume
          </a>
        </li>
        <li>
          <a
            href="/projects"
            target="_blank"
            title="My open source projects"
          >
            Open source
            <svg
              class="c-svg-external-link"
            >
              <use
                xlink:href="#svg-external-link"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            class=""
            href="/music"
            title="My music"
          >
            Music
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`NavComponent component Should unmount the NavComponent component 1`] = `<DocumentFragment />`;
